lib/ansible_tower_client/base_models/project.rb

```diff
@@ -19,8 +19,19 @@ def update
     end
 
     def last_update
-      return if related['last_update'].blank?
-      api.project_updates.find(related['last_update'])
+      return @last_update if defined? @last_update
+      return @last_update = nil unless related.raw_hash.key?('last_update')
+      return @last_update = nil if (update_id = related.last_update).blank?
+
+      unless update_id.kind_of?(Numeric) || update_id =~ /\A\d+\z/
+        if raw_hash.key?('summary_fields') && summary_fields.raw_hash.key?('last_update')
+          update_id = summary_fields.last_update.id
+        else
+          /\/(?'update_id'\d+)\/?\z/ =~ update_id
```

Comment:
Do you have an example of this string?  It might be nice to have as a comment or a test

Reply:
This is simple named group capture of all digits after the last `/`, optionally ending on the very last `/`.
E.g., this is url like `http://abc.def/something/12345/`

Reply:
Yes, I can mentally parse the regex, but it is only a small part of a fairly complicated method, which really should have tests.  My point is that it's much easier to add tests for a feature / method when it is being written that describe the expected inputs and outputs than for the developer fixing a bug to have to discover (or guess) all of the cases that need to be covered.